src/NNCache.h

```diff
@@ -65,6 +65,10 @@ class NNCache {
     }
 
     void dump_stats();
+    void clear_cache() {
+        m_cache.clear();
```

Comment:
If `m_order` isn't cleared then will the size check in `insert()` try erasing elements that have been cleared? If a hash is inserted that had previously been "cleared" then on a later insert that tries to erase the cleared hash, will that incorrectly remove all the elements equal to hash?

Reply:
Actually, would it be reasonable to include the komi value in the hash used for cache lookup thus removing the need to clear the cache?

Reply:
I think there is a problem for the example case I tried to explain above. I.e. when m_order contains a hash multiple times because it hasn’t been cleared and therefore when the size limit is hit (in m_order), m_cache will incorrectly have an entry removed.
There might be some speed benefits of maintaining the cache but like you, I’m not sure how to modify the hash function :)

Reply:
OK, I see the situation you meant now and I think not clearing `m_order` is a mistake. But now I also realize that not clearing the cache can save at least half of the NN evals if --pos or --neg is enabled, so I should indeed looking into including komi in hash!